firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/PatchMutation.java

```diff
@@ -146,27 +146,53 @@ public void applyToLocalView(MutableDocument document, Timestamp localWriteTime)
   @Override
   public Mutation squash(
       Mutation baseMutation, MutableDocument document, Timestamp localWriteTime) {
-    if (!getPrecondition().isValidFor(document)) {
+    MutableDocument preconditionTest =
+        (baseMutation != null && (baseMutation instanceof DeleteMutation))
```

Comment:
Is there some way we can make baseMutation non-nullable? We could use a VerifyMutation that does not verify anything for example, an empty merge with an exist precondition or a new EmptyMutation class. Ideally, this would allow us to always have the correct state in "document" which would mean that we do not have to merge preconditions.